source/agent/conference/roomController.js

```diff
@@ -350,12 +351,12 @@ module.exports.create = function (spec, on_init_ok, on_init_failed) {
         streams = {};
     };
 
-    var newTerminal = function (terminal_id, terminal_type, owner, preAssignedNode, on_ok, on_error) {
-        log.debug('newTerminal:', terminal_id, 'terminal_type:', terminal_type, 'owner:', owner);
+    var newTerminal = function (terminal_id, terminal_type, owner, preAssignedNode, origin, on_ok, on_error) {
+        log.debug('newTerminal:', terminal_id, 'terminal_type:', terminal_type, 'owner:', owner, " origin:", origin);
         if (terminals[terminal_id] === undefined) {
                 var purpose = (terminal_type === 'vmixer' || terminal_type === 'vxcoder') ? 'video'
                               : ((terminal_type === 'amixer' || terminal_type === 'axcoder') ? 'audio' : 'unknown');
-
+                mediaPreference.origin = origin;
                 var nodeLocality = (preAssignedNode ? Promise.resolve(preAssignedNode)
```

Comment:
Why not save `origin` in line:366 before `newTerminal` returns if you associate `origin` with `terminal`?